homeassistant/components/zwave_js/sensor.py

```diff
@@ -230,14 +234,72 @@ def unit_of_measurement(self) -> str | None:
 
         return str(self.info.primary_value.metadata.unit)
 
+
+class ZWaveMeterSensor(ZWaveNumericSensor, RestoreEntity):
+    """Representation of a Z-Wave Meter CC sensor."""
+
+    _attr_state_class = STATE_CLASS_MEASUREMENT
+
+    def __init__(
+        self,
+        config_entry: ConfigEntry,
+        client: ZwaveClient,
+        info: ZwaveDiscoveryInfo,
+    ) -> None:
+        """Initialize a ZWaveNumericSensor entity."""
+        super().__init__(config_entry, client, info)
+
+        # Entity class attributes
+        self._attr_last_reset = dt.utc_from_timestamp(0)
+        self._attr_device_class = DEVICE_CLASS_POWER
+        if self.info.primary_value.metadata.unit == "kWh":
+            self._attr_device_class = DEVICE_CLASS_ENERGY
+
+    async def async_update_last_reset(
+        self, node: ZwaveNode, endpoint: int, meter_type: int | None
+    ) -> None:
+        """Update last reset."""
+        # If the signal is not for this node or is for a different endpoint, ignore it
+        if self.info.node != node or self.info.primary_value.endpoint != endpoint:
+            return
+        # If a meter type was specified and doesn't match this entity's meter type,
+        # ignore it
+        if (
+            meter_type is not None
+            and self.info.primary_value.metadata.cc_specific.get("meterType")
+            != meter_type
+        ):
+            return
+
+        self._attr_last_reset = dt.utcnow()
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self) -> None:
+        """Call when entity is added."""
+        await super().async_added_to_hass()
+
+        # Restore the last reset time from stored state
+        restored_state = await self.async_get_last_state()
+        if restored_state and ATTR_LAST_RESET in restored_state.attributes:
+            self._attr_last_reset = dt.parse_datetime(
+                restored_state.attributes[ATTR_LAST_RESET]
+            )
+            self.async_write_ha_state()
```

Comment:
At the end of entity addition the entity state is always written so we don't need to do that explicitly here.
```suggestion
```